assimilation_code/modules/assimilation/adaptive_inflate_mod.f90

```diff
@@ -1200,23 +1262,27 @@ subroutine set_from_string(from_restart, single_file, from_string)
 
 end subroutine set_from_string
 
-!-----------------------------------------------------------------------
+
+!-------------------------------------------------------------------------------
+!>
 
 function nvalues_to_log(inflation_handle, from_string)
 type(adaptive_inflate_type), intent(in) :: inflation_handle
 character(len=*),            intent(in) :: from_string
 integer :: nvalues_to_log
 
-if ((inflation_handle%inflation_flavor == 3) .or. &
-    (inflation_handle%inflation_flavor == 2 .and. from_string == 'from namelist')) then
+if ((inflation_handle%inflation_flavor == SINGLE_SS_INFLATION) .or. &
```

Comment:
by the way, changing the magic numbers in this code to parameters makes it all much clearer.  good change.